### PR TITLE
wrap aws-sdk errors

### LINKED ISSUE #10574: Backstage container crash reboots when clicked on techdocs
## Expected Behavior

Published techdocs are displayed.

## Actual Behavior

The backstage container (not the pod) crash restarts.

## Steps to Reproduce

1. Click on Docs from the menu on the left.
2. Click on a component whose documentation you'd like to see.

## Context

- Techdocs are generated and published to S3 compatible object storage by a CI/CD pipeline. 
- The documentation shows up correctly when a component is selected and Docs tab is clicked.
- Log data 
```
2022-04-01T13:20:42.300Z backstage info ::ffff:10.42.0.29 - - [01/Apr/2022:13:20:42 +0000] "GET /api/search/query?term=&filters%5Bkind%5D=component&filters%5Bnamespace%5D=default&filters%5Bname%5D=backstage&pageCursor=&types%5B0%5D=techdocs HTTP/1.1" 200 14 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/100.0.4896.60 Safari/537.36" type=incomingRequest
/app/node_modules/@backstage/errors/dist/index.cjs.js:29
    throw new Error(`Encountered error object without a message, got '${value}'`);
          ^

Error: Encountered error object without a message, got 'NoSuchKey: null'
    at Object.assertError (/app/node_modules/@backstage/errors/dist/index.cjs.js:29:11)
    at /app/node_modules/@backstage/plugin-techdocs-node/dist/index.cjs.js:770:18
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

## Your Environment

Backstage is deployed in a kubernetes cluster.
Requests:{"cpu":"500m","memory":"2Gi"}
Limits:{"cpu":"8","memory":"2Gi"}

- Browser Information: 
Google Chrome - Version 100.0.4896.60 (Official Build) (x86_64)

- Output of `yarn backstage-cli info`: 
```
OS:   Darwin 21.1.0 - darwin/x64
node: v14.17.6
yarn: 1.22.17
cli:  0.16.0 (installed)

Dependencies:
  @backstage/app-defaults                            1.0.0
  @backstage/backend-common                          0.13.1
  @backstage/backend-tasks                           0.2.1
  @backstage/catalog-client                          1.0.0
  @backstage/catalog-model                           1.0.0
  @backstage/cli-common                              0.1.8
  @backstage/cli                                     0.16.0
  @backstage/config-loader                           1.0.0
  @backstage/config                                  1.0.0
  @backstage/core-app-api                            1.0.0
  @backstage/core-components                         0.9.2
  @backstage/core-plugin-api                         1.0.0
  @backstage/errors                                  1.0.0
  @backstage/integration-react                       1.0.0
  @backstage/integration                             1.0.0
  @backstage/plugin-api-docs                         0.8.3
  @backstage/plugin-app-backend                      0.3.30
  @backstage/plugin-auth-backend                     0.12.3
  @backstage/plugin-auth-node                        0.1.6
  @backstage/plugin-catalog-backend-module-bitbucket 0.1.1
  @backstage/plugin-catalog-backend-module-ldap      0.4.1
  @backstage/plugin-catalog-backend                  1.0.0
  @backstage/plugin-catalog-common                   1.0.0
  @backstage/plugin-catalog-graph                    0.2.15
  @backstage/plugin-catalog-import                   0.8.6
  @backstage/plugin-catalog-react                    1.0.0
  @backstage/plugin-catalog                          1.0.0
  @backstage/plugin-github-actions                   0.5.3
  @backstage/plugin-org                              0.5.3
  @backstage/plugin-permission-common                0.5.3
  @backstage/plugin-permission-node                  0.5.5
  @backstage/plugin-permission-react                 0.3.4
  @backstage/plugin-proxy-backend                    0.2.24
  @backstage/plugin-scaffolder-backend               1.0.0
  @backstage/plugin-scaffolder-common                1.0.0
  @backstage/plugin-scaffolder                       1.0.0
  @backstage/plugin-search-backend-node              0.5.2
  @backstage/plugin-search-backend                   0.4.8
  @backstage/plugin-search-common                    0.3.2
  @backstage/plugin-search                           0.7.4
  @backstage/plugin-sonarqube                        0.3.3
  @backstage/plugin-tech-radar                       0.5.10
  @backstage/plugin-techdocs-backend                 1.0.0
  @backstage/plugin-techdocs-node                    1.0.0
  @backstage/plugin-techdocs                         1.0.0
  @backstage/plugin-user-settings                    0.4.2
  @backstage/release-manifests                       0.0.2
  @backstage/search-common                           0.3.2
  @backstage/test-utils                              1.0.0
  @backstage/theme                                   0.2.15
  @backstage/types                                   1.0.0
  @backstage/version-bridge                          1.0.0
```

### PR DESCRIPTION
**Pull request by manuelstein**
## Hey, I just made a Pull Request!

Resolves #10574 - the NoSuchKey: null error mentioned in the bug is actually an HTTP error response thrown by the aws-sdk with no message.

When techdocs tries to access S3, there may be http errors thrown with no message.
Those errors do not conform to the @backstage/errors' assertError check.
As a resolution, any error reading the stream is wrapped in a ForwardError.
The test case was updated accordingly.
#### :heavy_check_mark: Checklist

- [X] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [X] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

Signed-off-by: Manuel Stein <manuel.stein@nokia-bell-labs.com>

### CODE REVIEW

**Review by benjdlambert — APPROVED** [review]
Nice! Thanks for this! :pray:

**Comment by benjdlambert** [comment]
Think you need to run `prettier` on the changed files.

```
yarn prettier --write .changeset/loud-panthers-arrive.md plugins/techdocs-node/src/stages/publish/awsS3.ts
```

And commit the changes :)

**Comment by manuelstein** [comment]
Sorry, my vscode prettier run didn't pick up the repo's prettier config. Just updated the PR

File: plugins/techdocs-node/src/stages/publish/awsS3.test.ts [hunk 1]
```diff
@@ -478,7 +478,7 @@ describe('AwsS3Publish', () => {
 
       await expect(fails).rejects.toMatchObject({
         message: expect.stringMatching(
-          /TechDocs metadata fetch failed; caused by Error: The file .* does not exist/i,
```
Comment (benjdlambert) [inline plugins/techdocs-node/src/stages/publish/awsS3.test.ts:481]: This looks a little strange that we're closing the scope for the regex when we're adding the ability to log more info. Is there a way that we can include the error message in this regex?

**Review by benjdlambert — COMMENTED** [review]
Just a quick comment

File: plugins/techdocs-node/src/stages/publish/awsS3.test.ts (hunk 1 above)
Comment (manuelstein) [reply plugins/techdocs-node/src/stages/publish/awsS3.test.ts:481]: Sure, the error is from the aws-sdk getObject mockup in [L64](https://github.com/nokia/backstage/blob/6c118dd093eea6f96004bb8ee3a48a8391dcdb8d/plugins/techdocs-node/src/stages/publish/awsS3.test.ts#L62-L65), so we can pin it to the exact error message.

Comment (manuelstein) [reply plugins/techdocs-node/src/stages/publish/awsS3.test.ts:481]: no regex needed

**Review by camilaibs — APPROVED** [review]
Thanks for contributing 🎉

**Pull request merged**